nebula_examples/launch/hesai_offline_bag_pcd.xml

```diff
@@ -1,25 +1,41 @@
 <?xml version="1.0"?>
 <launch>
-    <arg name="sensor_model" description="Pandar64|Pandar40P|PandarXT32|PandarXT32M|PandarAT128|PandarQT64"/>
-    <arg name="return_mode" default="Dual" description="See readme for supported return modes"/>
-    <arg name="frame_id"  default="hesai"/>
+    <!-- action selection -->
+    <arg name="is_write_to_pcd" default="false"/>
+    <arg name="is_write_to_rosbag" default="true"/>
+    <arg name="is_write_packets_to_rosbag" default="false"/>
+
+    <!-- sensor related configuration -->
+    <arg name="sensor_model" default="Pandar128E4X" description="Pandar64|Pandar40P|PandarXT32|PandarXT32M|PandarAT128|PandarQT64"/>
+    <arg name="lidar_parameter_file" default="$(find-pkg-share nebula_ros)/config/lidar/hesai/Pandar128E4X.params.yaml"/>
+    <arg name="bag_path" default="rosbags/default_2024-10-24-10-50-48_p0900"/>
+    <arg name="target_topic" default="/sensing/lidar/top/pandar_packets"/>
+    <arg name="output_topic" default="/sensing/lidar/top/pointcloud"/>
+    <arg name="frame_id"  default="lidar_top"/>
+    <arg name="out_path" default="output_rosbag"/>
+
+    <!-- output sample number configuration -->
+    <arg name="out_num" default="5"/>
+    <arg name="skip_num" default="1"/>
+
+    <!-- LiDAR configuration overriding parameter file -->
     <arg name="scan_phase" default="0.0" />
+    <arg name="return_mode" default="Dual" description="See readme for supported return modes"/>
 
     <arg name="calibration_file" default="$(find-pkg-share nebula_decoders)/calibration/hesai/$(var sensor_model).csv"/>
     <arg name="correction_file" default="$(find-pkg-share nebula_decoders)/calibration/hesai/$(var sensor_model).dat"/>
 
-    <arg name="bag_path" default="path to bag dir"/>
     <arg name="storage_id" default="sqlite3"/>
-    <arg name="out_path" default="path to output dir"/>
     <arg name="format" default="cdr"/>
-    <arg name="target_topic" default="/pandar_packets"/>
-    <arg name="out_num" default="1"/>
-    <arg name="skip_num" default="3"/>
     <arg name="only_xyz" default="true"/>
 
 
     <node pkg="nebula_examples" exec="hesai_ros_offline_extract_bag_pcd_node"
           name="hesai_cloud" output="screen">
+        <param from="$(var lidar_parameter_file)"/>
+        <param name="is_write_to_pcd" value="$(var is_write_to_pcd)"/>
```

Comment:
Allow substitutions so that variables in the param file get resolved (e.g. calibration_file)
```suggestion
        <param from="$(var lidar_parameter_file)" allow_substs="true" />
```